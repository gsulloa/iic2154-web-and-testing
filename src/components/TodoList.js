import React from "react";

const List = props => (
  <ol>
    {props.data.map(element => {
      return <li key={element.id}>{element.text}</li>;
    })}
  </ol>
);

export class TodoForm extends React.Component {
  state = {
    text: ""
  };
  handleWrite = e => {
    this.setState({ text: e.target.value });
  };
  render() {
    return (
      <form
        onSubmit={e => {
          this.props.handleSubmit(e, this.state.text);
          this.setState({ text: "" });
        }}
      >
        <label htmlFor="text">Ingresa un elemento</label>
        <input onChange={this.handleWrite} id="text" value={this.state.text} />
      </form>
    );
  }
}

class TodoList extends React.Component {
  state = {
    list: []
  };
  handleSubmit = (e, text) => {
    e.preventDefault();
    this.setState({
      list: [...this.state.list, { id: this.state.list.length, text: text }],
      text: ""
    });
  };
  render() {
    return (
      <div>
        <h1>TodoList</h1>
        <TodoForm handleSubmit={this.handleSubmit} />
        <List data={this.state.list} />
      </div>
    );
  }
}

export default TodoList;
