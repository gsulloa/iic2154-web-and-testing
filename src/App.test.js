import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import App from "./App";
import TodoList from "./components/TodoList";

it("renders without crashing", () => {
  const { container, getByText, queryByText, debug } = render(<App />);
  expect(getByText("TodoList")).toBeInTheDocument();
  expect(queryByText("No existe")).toBeNull();
  expect(container.firstChild).toMatchSnapshot();
});

it("todolist", () => {
  const { container, getByLabelText, getByText } = render(<TodoList />);
  expect(container.firstChild).toMatchSnapshot();
  const input = getByLabelText(/ingresa un elemento/i);
  fireEvent.change(input, { target: { value: "Texto nuevo" } });
  expect(input).toHaveValue("Texto nuevo");
  expect(container.firstChild).toMatchSnapshot();
  fireEvent.submit(input);
  expect(container.firstChild).toMatchSnapshot();
  expect(getByText("Texto nuevo")).toBeInTheDocument();
});
