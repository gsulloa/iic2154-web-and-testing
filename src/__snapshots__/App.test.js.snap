// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <h1>
    TodoList
  </h1>
  <form>
    <label
      for="text"
    >
      Ingresa un elemento
    </label>
    <input
      id="text"
      value=""
    />
  </form>
  <ol />
</div>
`;

exports[`todolist 1`] = `
<div>
  <h1>
    TodoList
  </h1>
  <form>
    <label
      for="text"
    >
      Ingresa un elemento
    </label>
    <input
      id="text"
      value=""
    />
  </form>
  <ol />
</div>
`;

exports[`todolist 2`] = `
<div>
  <h1>
    TodoList
  </h1>
  <form>
    <label
      for="text"
    >
      Ingresa un elemento
    </label>
    <input
      id="text"
      value="Texto nuevo"
    />
  </form>
  <ol />
</div>
`;

exports[`todolist 3`] = `
<div>
  <h1>
    TodoList
  </h1>
  <form>
    <label
      for="text"
    >
      Ingresa un elemento
    </label>
    <input
      id="text"
      value=""
    />
  </form>
  <ol>
    <li>
      Texto nuevo
    </li>
  </ol>
</div>
`;
